### DISCUSSION TITLE
Approach for binary command completion to show TLDR Pages (or man pages)

### DISCUSSION THREAD

**Discussion by jeeftor** [junegunn/fzf#3825] [Q&A]
I'm trying to understand the best way to go about this and/or if there is a function that can already handle it.

I'm using `zsh` and thought _Wouldn't it be cool if I typed a command the preview showed TLDR pages for it_

Is there a function I can call for doing "command" completion?

I was messing with stuff and it didn't seem to pass the command I'm trying to complete into my preview function...

![image](https://github.com/junegunn/fzf/assets/6491743/f8daada6-a4c1-424a-9100-6681dadbf4a5)

```bash
echo "ARGS [$@]"
```

I'm sure I'm missing something somewhere

```zsh

# Disable menu completions for everything
zstyle ':completion:*' menu no

# Set Custom fzf Preview for All Completions - using fzf-preview
zstyle ':fzf-tab:complete:*' fzf-preview '$HOME/bin/fzf-preview.sh $realpath'
zstyle ':fzf-tab:complete:*' fzf-bindings 'ctrl-/:toggle-preview'

# Set Custom fzf Preview for cd Command
zstyle ':fzf-tab:complete:cd:*' fzf-preview 'lsd --tree --color=always  --git $realpath | head -200'

# Disable fzf Preview for git
zstyle ':fzf-tab:complete:git:*' fzf-preview ''

# Configure Keys to Switch Completion Groups in fzf
zstyle ':fzf-tab:*' switch-group '<' '>'
```

**Comment by LangLangBart** [comment]
> echo "ARGS [$@]"

If you would like to reference some value you could use placeholders or some environment variables exported by `fzf`.

### Placeholder

|     Placeholder      |            Description            |
| :------------------: | :-------------------------------: |
|         `{}`         |           current line            |
| `{2}` |  current line with positional field 2  |
| `{2..}` |  current line from the 2nd field to the last field |
|        `{n}`         | current line number (zero index)  |
| `{q}`  |           current query           |


| Placeholder expression flags |                Description                 |
| :--------------------------: | :----------------------------------------: |
|             `+`              | space-separated list of the selected lines |
|             `f`              |               temporary file               |
|             `s`              |            preserve whitespace             |

### Environment Variables

You can skim the `fzf` man page or view them in the preview window.

```sh
man fzf | less --pattern "ENVIRONMENT VARIABLES EXPORTED TO CHILD PROCESSES"
```

```sh
: | fzf --preview "typeset | grep '^FZF_'"
```

> I'm using zsh and thought Wouldn't it be cool if I typed a command the preview showed TLDR pages for it

The wiki has some examples, e.g.:

- [junegunn/fzf Wiki · fzf-man-pages widget (for zsh)](https://github.com/junegunn/fzf/wiki/Examples#fzf-man-pages-widget-for-zsh)

**Reply by jeeftor** [reply]
I found the desc env var had what I wanted. Which is odd
